File: LeetCode/1400. Construct K Palindrome Strings/solution.py
class Solution:
    # def canConstruct(self, s: str, k: int) -> bool:
    #     cc = Counter(s)
    #     return sum(map(lambda x:x&1,cc.values())) <= k and len(s) >= k
    
    def canConstruct(self, s, k):
        return sum(i & 1 for i in collections.Counter(s).values()) <= k <= len(s)

File: LeetCode/1400. Construct K Palindrome Strings/solution.cpp
class Solution {
public:
    bool canConstruct(string s, int k) {
        int odds = 0, len = s.size();
        vector<int> cnt(26,0);
        for (auto ch : s)
            cnt[ch-'a']++;
        for (int i = 0; i < 26; i++){
            if (cnt[i]&1)
                odds++;
        }
        if (odds > k || len < k)
            return false;
        return true;
    }
};